src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx

```diff
@@ -0,0 +1,210 @@
+
+type CreateXYKPoolFormProps = {
+  assetA: string
+  assetB: string
+  onAssetAOpen: () => void
+  onAssetBOpen: () => void
+  onClose: () => void
+}
+
+export const CreateXYKPoolForm = ({
+  assetA,
+  assetB,
+  onClose,
+  onAssetAOpen,
+  onAssetBOpen,
+}: CreateXYKPoolFormProps) => {
+  const { t } = useTranslation()
+  const queryClient = useQueryClient()
+
+  const { api, assets } = useRpcProvider()
+
+  const assetAMeta = assets.getAsset(assetA ?? "")
+  const assetBMeta = assets.getAsset(assetB ?? "")
+
+  const { account } = useAccount()
+
+  const { data: balanceA } = useTokenBalance(assetA, account?.address)
+  const { data: balanceB } = useTokenBalance(assetB, account?.address)
+
+  const form = useForm<CreateXYKPoolFormData>({
+    mode: "onChange",
+    resolver: zodResolver(
+      createXYKPoolFormSchema(
+        balanceA?.balance ?? BN_0,
+        assetAMeta.decimals,
+        balanceB?.balance ?? BN_0,
+        assetBMeta.decimals,
+      ),
+    ),
+    defaultValues: {
+      assetA: "",
+      assetB: "",
+    },
+  })
+
+  const assetAValue = BigNumber(form.watch("assetA"))
+  const assetBValue = BigNumber(form.watch("assetB"))
+
+  const { createTransaction } = useStore()
+
+  const handleSubmit = async (values: CreateXYKPoolFormData) => {
+    const data = {
+      assetA: {
+        id: assetAMeta.id,
+        amount: new BigNumber(values.assetA)
+          .shiftedBy(assetAMeta.decimals)
+          .toFixed(),
+      },
+      assetB: {
+        id: assetBMeta.id,
+        amount: new BigNumber(values.assetB)
+          .shiftedBy(assetBMeta.decimals)
+          .toFixed(),
+      },
+    }
+
+    const toast = TOAST_MESSAGES.reduce((memo, type) => {
+      const msType = type === "onError" ? "onLoading" : type
+      memo[type] = (
+        <Trans
+          t={t}
+          i18nKey={`liquidity.pool.xyk.create.toast.${msType}`}
+          tOptions={{
+            symbolA: assetAMeta.symbol,
+            symbolB: assetBMeta.symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      )
+      return memo
+    }, {} as ToastMessage)
+
+    await createTransaction(
+      {
+        tx: api.tx.xyk.createPool(
+          data.assetA.id,
+          data.assetA.amount,
+          data.assetB.id,
+          data.assetB.amount,
+        ),
+      },
+      {
+        onClose,
+        onBack: () => {},
+        onSubmitted: () => {
+          onClose()
+          form.reset()
+        },
+        onSuccess: () => {
+          queryClient.refetchQueries(QUERY_KEYS.xykPools)
+        },
+        toast,
+      },
+    )
+  }
+
+  return (
+    <form
+      onSubmit={form.handleSubmit(handleSubmit)}
+      autoComplete="off"
+      sx={{
+        flex: "column",
+        justify: "space-between",
+        minHeight: "100%",
+      }}
+    >
+      <Controller
+        name="assetA"
+        control={form.control}
+        rules={{
```

Comment:
why do you use rules prop if you are using zod?

Reply:
forgot to remove, will fix